irods_testing_environment/irods_config.py

```diff
@@ -436,11 +446,31 @@ def configure_irods_federation_testing(ctx, remote_zone, zone_where_tests_will_r
     # add specific query to the local zone
     bug_3466_query = 'select alias, sqlStr from R_SPECIFIC_QUERY'
     asq = 'iadmin asq \'{}\' {}'.format(bug_3466_query, 'bug_3466_query')
-    logging.info('creating specific query[{}] [{}]'.format(asq, container.name))
+    logging.info('creating specific query [{}] [{}]'.format(asq, container.name))
     if execute.execute_command(container, asq, user='irods') is not 0:
         raise RuntimeError('failed to create specific query [{}] [{}]'
                            .format(bug_3466_query, container.name))
 
+    # reload configuration for both zones if iRODS 5 is installed
+    if server_version_is_irods_5(container):
+        reload_config = "python3 -c 'from scripts.irods.controller import IrodsController; IrodsController().reload_configuration()'"
+        logging.info('reloading configuration for [{}] [{}]'.format(reload_config, container.name))
+        if execute.execute_command(container, reload_config, user='irods', workdir=context.irods_home()) is not 0:
+            raise RuntimeError('failed to reload configuration [{}] [{}]'
+                               .format(reload_config, container.name))
+
+        container = ctx.docker_client.containers.get(
+            context.irods_catalog_provider_container(
+                ctx.compose_project.name,
+                service_instance=zone_where_tests_will_run.provider_service_instance
+            )
+        )
+
+        logging.info('reloading configuration for [{}] [{}]'.format(reload_config, container.name))
+        if execute.execute_command(container, reload_config, user='irods', workdir=context.irods_home()) is not 0:
+            raise RuntimeError('failed to reload configuration [{}] [{}]'
+                               .format(reload_config, container.name))
+
```

Comment:
What if the remote zone is not running iRODS 5 but this zone is running iRODS 5, or vice versa? Maybe a moot point if we never add federation testing for different major versions before 4.x is EOL'd...

Reply:
Given we don't currently have a way to express that (easily, I think), I don't think it applies to this PR.

However, I think we're going to have to expand the testing environment to support that eventually. Do we have an issue for that?

Reply:
I created https://github.com/irods/irods_testing_environment/issues/240 for mixed-version federation. Resolving.